naming/src/main/java/org/jboss/as/naming/NamingStore.java

```diff
@@ -86,6 +86,16 @@
     List<Binding> listBindings(Name name) throws NamingException;
 
     /**
+     * Retrieves the named object, following links except for the terminal atomic component of the name.
+     * If the object bound to <tt>name</tt> is not a link, returns the object itself.
+     *
+     * @param name The entry name
+     * @return The link from the store.
+     * @throws NamingException If any errors occur.
+     */
+    Object lookupLink(Name name) throws NamingException;
```

Comment:
I don't see why this needs a special method on the naming store, as the naming context already handles LinkRef lookup logic